### PR TITLE
Weekly digest API endpoints

### PR DESCRIPTION
**Pull request by jtorreggiani**
- [X] Create weekly digest app
- [X] Create WeeklyDigestSubscription data model
- [X] Creates CRUD Django rest framework views 
- [X] Add app to `config/settings.py` file
- [X] Add API tests

Commands used to scaffold the weekly digest app and data models.

```Shell
docker compose python manage.py startapp weekly_digest
docker compose python manage.py makemigrations
docker compose python manage.py migrate
```

### CODE REVIEW

**Comment by jtorreggiani** [comment]
@dashcraft and @jackwkinsey, put this PR together based on your feedback in https://github.com/data-umbrella/event-board-api/issues/59. I've set up the weekly digest API to support all the CRUD endpoints, but in practice we may only need create and update or destroy. We can remove the unused API endpoints, but this should get us started.

File: config/settings.py [hunk 1]
```diff
@@ -74,6 +74,7 @@
     'authentication',
     'storages',
     'contact_emails',
+    'weekly_digest',
```
Comment (jtorreggiani) [inline config/settings.py:77]: If you don't include the app here the command `python manage.py makemigrations` won't pick up new models.

File: config/urls.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
     path('api/v1/', include('events.urls')),
     path('api/v1/', include('authentication.urls')),
     path('api/v1/', include('contact_emails.urls')),
+    path('api/v1/', include('weekly_digest.urls')),
```
Comment (jtorreggiani) [inline config/urls.py:27]: Need to include the urls for the weekly digest app here.

File: weekly_digest/admin.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (jtorreggiani) [inline weekly_digest/admin.py:3]: If we think the admin will need me able to access the subscriptions via the Django admin we need to register the WeeklyDigestSubscription model here.

**Comment by jackwkinsey** [comment]
@jtorreggiani 

Should an endpoint be included for getting all of the emails that are subscribed? I assume the cronjob will need to get a list of emails that are subscribed to the weekly digest, build the email by adding the events for the week, and then send the emails out to the list received.

**Review by jackwkinsey — APPROVED** [review]
Looks great! Also serves as a great example for adding further functionality in the future. 👍

**Pull request merged**